dashboard/test/models/script_test.rb

```diff
@@ -2141,501 +1857,6 @@ def populate_cache_and_disconnect_db
-    assert_equal 1, unit.script_levels[3].position
-    assert_equal 5, unit.script_levels[4].chapter
-    assert_equal 2, unit.script_levels[4].position
-  end
-
-  test 'can add lesson with no levels' do
-    dsl = <<-UNIT
-      lesson_group 'content', display_name: 'Content'
-      lesson 'Lesson1', display_name: 'Lesson1'
-
-    UNIT
-
-    unit = Script.add_unit(
-      {name: 'lesson-group-test-script'},
-      ScriptDSL.parse(dsl, 'a filename')[0][:lesson_groups]
-    )
-    lesson = unit.lessons.first
-    assert_equal 'Lesson1', lesson.key
-    assert_equal 0, lesson.script_levels.count
-  end
-
-  test 'raise error if try to change key of lesson in stable and i18n unit' do
-    ScriptConstants.stubs(:i18n?).with('coursea-2017').returns(true)
-
-    new_dsl = <<-UNIT
-      published_state 'stable'
-
-      lesson 'Debugging: Unspotted Bugs 1', display_name: 'Debugging: Unspotted Bugs'
-      level 'courseB_video_Unspotted'
-    UNIT
-
-    raise = assert_raises do
-      Script.add_unit(
-        {name: 'coursea-2017'},
-        ScriptDSL.parse(new_dsl, 'a filename')[0][:lesson_groups]
-      )
-    end
-    assert_equal 'Adding new keys or update existing keys for lessons in units that are marked as stable and included in the i18n sync is not allowed. Offending Lesson Key: Debugging: Unspotted Bugs 1', raise.message
-  end
-
-  test 'raise error if try to add lesson in stable and i18n unit' do
-    ScriptConstants.stubs(:i18n?).with('coursea-2017').returns(true)
-
-    l1 = create :level
-
-    new_dsl = <<-UNIT
-      published_state 'stable'
-
-      lesson 'new-lesson', display_name: 'New Lesson'
-      level '#{l1.name}'
-
-      lesson 'Debugging: Unspotted Bugs 1', display_name: 'Debugging: Unspotted Bugs'
-      level 'courseB_video_Unspotted'
-    UNIT
-
-    raise = assert_raises do
-      Script.add_unit(
-        {name: 'coursea-2017'},
-        ScriptDSL.parse(new_dsl, 'a filename')[0][:lesson_groups]
-      )
-    end
-    assert_equal 'Adding new keys or update existing keys for lessons in units that are marked as stable and included in the i18n sync is not allowed. Offending Lesson Key: new-lesson', raise.message
-  end
-
-  test 'raise error if try to add new lesson group in stable and i18n unit' do
-    ScriptConstants.stubs(:i18n?).with('coursea-2017').returns(true)
-
-    new_dsl = <<-UNIT
-      published_state 'stable'
-
-      lesson_group 'lg', display_name: 'Lesson Group'
-      lesson 'Debugging: Unspotted Bugs 1', display_name: 'Debugging: Unspotted Bugs'
-      level 'courseB_video_Unspotted'
-    UNIT
-
-    raise = assert_raises do
-      Script.add_unit(
-        {name: 'coursea-2017'},
-        ScriptDSL.parse(new_dsl, 'a filename')[0][:lesson_groups]
-      )
-    end
-    assert_equal 'Adding new keys or update existing keys for lesson groups in scripts that are marked as stable and included in the i18n sync is not allowed. Offending Lesson Group Key: lg', raise.message
-  end
```

Comment:
this huge block of removed tests from lines 2114-2637 covered both `ScriptDSL.parse` and the branch of `Script.add_unit` which dealt with unmigrated units. because both of these codepaths are dead now, it seems ok to remove these tests. I did not go through and spot check to see if any of these cases are covered in our new codepath -- my thought there is that our new codepath for editing units and lessons has survived an entire year of editing across CSF, CSD, CSP, CSA and CSC, so I'm tempted to rely on that as evidence that our test coverage there is good, and add more test coverage there as problems arise.

that said, if there are specific test cases which I'm removing here which seem both missing from and important enough to add back to our migrated unit / lesson update tests, I'd be happy to add a few one-offs.

I am open to pushback here -- I am trying to find a good balance of quality and pragmatism.